layers/ip6_test.go

```diff
@@ -428,3 +428,115 @@ func TestIPv6JumbogramDecode(t *testing.T) {
 		t.Error("No Payload layer type found in packet")
 	}
 }
+
+var testPacketIPv6Routing = []byte{
+	0x60, 0x00, 0x00, 0x00, 0x00, 0x38, 0x2b, 0x40, 0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3b, 0x06, 0x04, 0x02, 0x03, 0x00, 0x00, 0x00,
```

Comment:
```suggestion
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3b, 0x06, 0x04, 0x02, 0x02, 0x00, 0x00, 0x00,
```
- SourceRoutingIPs[0] = 2001:db8::2
- SourceRoutingIPs[1] = 2001:db8::3
- SourceRoutingIPs[2] = 2001:db8::4
Therefore, the index of the last entry is 2 instead of 3